packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -443,6 +460,42 @@ private void setImageStreamImageAvailableListener(final EventChannel.EventSink i
         null);
   }
 
+  // Torch
+  public void setTorchMode(@NonNull final Result result, boolean enable) {
+    setTorchMode(result, enable, 1.0);
+  }
+
+  public void setTorchMode(@NonNull final Result result, boolean enable, double level) {
+    try {
+      captureRequestBuilder.set(
+          CaptureRequest.FLASH_MODE,
+          enable ? CaptureRequest.FLASH_MODE_TORCH : CaptureRequest.FLASH_MODE_OFF);
+
+      captureRequestBuilder.set(
```

Comment:
Why are we always setting this now no matter the `FLASH_MODE`?

Reply:
This is how it works with "android.hardware.camera2" in the last Android SDK.
To change the state of the flash, we need to change the value of [FLASH_MODE](https://developer.android.com/reference/android/hardware/camera2/CaptureRequest.html#FLASH_MODE) through the [CaptureRequest](https://developer.android.com/reference/android/hardware/camera2/CaptureRequest.html).